gemini/use-cases/document-rag-on-llamaindex/llamaindex_rag.ipynb

```diff
@@ -0,0 +1,1458 @@
+        "        index.resource_name\n",
+        "        for index in aiplatform.MatchingEngineIndex.list(\n",
+        "            filter=f\"display_name={index_name}\"\n",
+        "        )\n",
+        "    ]\n",
+        "\n",
+        "    if len(index_names) == 0:\n",
+        "        print(f\"Creating Vector Search index {index_name} ...\")\n",
+        "        vs_index = aiplatform.MatchingEngineIndex.create_tree_ah_index(\n",
+        "            display_name=index_name,\n",
+        "            dimensions=index_dimensions,\n",
+        "            distance_measure_type=\"DOT_PRODUCT_DISTANCE\",\n",
+        "            shard_size=\"SHARD_SIZE_SMALL\",\n",
+        "            index_update_method=\"STREAM_UPDATE\",  # allowed values BATCH_UPDATE , STREAM_UPDATE,\n",
+        "            approximate_neighbors_count=5,\n",
+        "        )\n",
+        "        print(\n",
+        "            f\"Vector Search index {vs_index.display_name} created with resource name {vs_index.resource_name}\"\n",
+        "        )\n",
+        "    else:\n",
+        "        vs_index = aiplatform.MatchingEngineIndex(index_name=index_names[0])\n",
+        "        print(\n",
+        "            f\"Vector Search index {vs_index.display_name} exists with resource name {vs_index.resource_name}\"\n",
+        "        )\n",
+        "\n",
+        "    return vs_index\n",
+        "\n",
+        "\n",
+        "def create_vertexai_search_endpoint(endpoint_name):\n",
+        "    \"\"\"\n",
+        "    Creates a Vertex AI Search endpoint.\n",
+        "    \"\"\"\n",
+        "    endpoint_names = [\n",
+        "        endpoint.resource_name\n",
+        "        for endpoint in aiplatform.MatchingEngineIndexEndpoint.list(\n",
+        "            filter=f\"display_name={endpoint_name}\"\n",
+        "        )\n",
+        "    ]\n",
+        "\n",
+        "    if len(endpoint_names) == 0:\n",
+        "        print(f\"Creating Vector Search index endpoint {endpoint_name} ...\")\n",
+        "        vs_endpoint = aiplatform.MatchingEngineIndexEndpoint.create(\n",
+        "            display_name=endpoint_name, public_endpoint_enabled=True\n",
+        "        )\n",
+        "        print(\n",
+        "            f\"Vector Search index endpoint {vs_endpoint.display_name} created with resource name {vs_endpoint.resource_name}\"\n",
+        "        )\n",
+        "    else:\n",
+        "        vs_endpoint = aiplatform.MatchingEngineIndexEndpoint(\n",
+        "            index_endpoint_name=endpoint_names[0]\n",
+        "        )\n",
+        "        print(\n",
+        "            f\"Vector Search index endpoint {vs_endpoint.display_name} exists with resource name {vs_endpoint.resource_name}\"\n",
+        "        )\n",
+        "\n",
+        "    return vs_endpoint\n",
+        "\n",
+        "\n",
+        "def deploy_vertexai_search_endpoint(vs_index, vs_endpoint, index_name):\n",
```

Comment:
Also, I'm not sure these extra wrapper methods need to be created. They're single function calls and the user can decide whether or not to run them based on if the endpoints are created already.

Reply:
If the endpoints are created already it will return the IDs. I added comments to say if you have existing endpoint and endpoint, update them in the global variables.

Resolving for right now, if you prefer a change let me know.